cpu/atmega_common/periph/spi.c

```diff
@@ -47,17 +49,28 @@ void spi_init(spi_t bus)
 void spi_init_pins(spi_t bus)
 {
 
-    /* the pin configuration for this CPU is fixed:
+    /* The pin configuration for atmega2560 and atmega1281 is:
      * - PB3: MISO (configure as input - done automatically)
      * - PB2: MOSI (configure as output)
      * - PB1: SCK  (configure as output)
      * - PB0: SS   (configure as output, but unused)
      *
+     * The pin configuration for atmega328p is:
+     * - PB4: MISO (configure as input - done automatically)
+     * - PB3: MOSI (configure as output)
+     * - PB5: SCK  (configure as output)
+     * - PB2: SS   (configure as output, but unused)
+     *
      * The SS pin must be configured as output for the SPI device to work as
      * master correctly, though we do not use it for now (as we handle the chip
      * select externally for now)
      */
-    DDRB |= ((1 << DDB2) | (1 << DDB1) | (1 << DDB0));
+     #if defined (CPU_ATMEGA2560) || defined (CPU_ATMEGA1281)
+     DDRB |= ((1 << DDB2) | (1 << DDB1) | (1 << DDB0));
+     #endif
+     #ifdef CPU_ATMEGA328P
+     DDRB |= ((1 << DDB2) | (1 << DDB3) | (1 << DDB5));
+     #endif
```

Comment:
Did you change this indentation? In my browser it appears without change but maybe is only my browser.